packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -417,8 +437,14 @@ function transform(root: IRNode, codeGen: CodeGen): t.Expression {
                 return t.stringLiteral(attr.value);
             }
 
-            case IRAttributeType.Boolean:
-                return t.booleanLiteral(attr.value);
+            case IRAttributeType.Boolean: {
+                // A boolean value used in an attribute should always generate .setAttribute(attr.name, ''),
+                // regardless if is a boolean attribute or not.
+                // Note: in Firefox, elm.setAttribute('spellcheck', ''), sets the attribute, but the property value is false.
+                return isUsedAsAttribute && attr.name !== 'spellcheck'
```

Comment:
The spellcheck attribute is enumerated and expects `true` or `false` as value. Setting it to `` is invalid and automatically fall back to the default user agent behavior: https://html.spec.whatwg.org/multipage/interaction.html#spelling-and-grammar-checking. The default user agent behavior might differ and it's fine.

IMO, we should not be in the business of fixing invalid HTML, and we should set the attribute value as even if we know it is wrong. A better approach would be to fail the compilation and report an error.

Reply:
mm, i see what you mean, basically `<div spellcheck></div>` is wrong in firefox anyways, we are not fixing it for you; makes sense, im good with it.

however, im not good adding a compilation error, at least not as part of this pr.